lib/Sema/CSWalker.cpp

```diff
@@ -325,7 +325,18 @@ class LocalCSWalker : public glu::ast::ASTWalker<LocalCSWalker, void> {
         );
     }
 
-    void postVisitStructInitializerExpr(glu::ast::StructInitializerExpr *) { }
+    void postVisitStructInitializerExpr(glu::ast::StructInitializerExpr *node)
+    {
+        auto *structType = node->getType();
+
+        if (!structType)
+            return;
+
+        auto constraint = Constraint::createStructInitialiser(
+            _cs.getAllocator(), structType, node
+        );
+        _cs.addConstraint(constraint);
```

Comment:
the type cannot possibly be null
```suggestion
        auto constraint = Constraint::createStructInitialiser(
            _cs.getAllocator(), structType, node->getType()
        );
        _cs.addConstraint(constraint);
```